docs/proton-kafka.md

```diff
@@ -559,19 +558,28 @@ CREATE EXTERNAL STREAM example (
     _tp_message_headers map(string, string)
 ) settings type='kafka',...;
 ```
-Then insert data to the external stream via `INSERT INTO` or materialized views, with a map of string pairs as custom headers for each message.
 
-## DROP EXTERNAL STREAM
+Then insert rows to the external stream via `INSERT INTO` or Materialized Views, the `_tp_message_headers` will be set to the headers of the Kafka message.
+
+#### sharding_expr (Partition Logic)
+
+`sharding_expr` is used to control how rows are distributed to Kafka partitions:
 
 ```sql
-DROP STREAM [IF EXISTS] stream_name
+CREATE EXTERNAL STREAM foo (
+    id int32,..
+) SETTINGS type='kafka', sharding_expr='hash(id)'...;
 ```
 
+When inserting rows, the partition ID will be evaluated based on the `sharding_expr` and Timeplus will put the message into the corresponding Kafka partition.
+
 ## Properties for Kafka client {#advanced_settings}
 
-For more advanced use cases, you can specify customized properties while creating the external streams. Those properties will be passed to the underlying Kafka client, which is [librdkafka](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md).
+In advanced use cases, you may want to fine-tune the behavior of the Kafka consumer, producer, or topic when creating Kafka external streams. Timeplus allows this through the `properties` setting, which passes configuration options directly to the underlying [librdkafka](https://github.com/confluentinc/librdkafka) client.
 
-For example:
+These settings can control aspects like message size limits, retry behavior, timeouts, and more. For a full list of available configuration options, refer to the [librdkafka configuration documentation](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md).
+
+### Example
```

Comment:
If you don't specify {#id}, docusaus framework will generate the url hash based on the header name. But there are 2 "Example" in the docs, so the second one will be "example-1", in this case, better to explicitly set the hash code